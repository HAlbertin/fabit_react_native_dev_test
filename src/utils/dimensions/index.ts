import { Dimensions } from 'react-native';

const deviceWidth = Dimensions.get('window').width;
const deviceHeight = Dimensions.get('window').height;

const DimensionUtils = {
  deviceWidth,
  deviceHeight,
};

export default DimensionUtils;
