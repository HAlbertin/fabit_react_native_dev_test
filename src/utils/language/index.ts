import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import StorageUtils from '../storage';
import de from './translations/de';
import en from './translations/en';
import es from './translations/es';
import fr from './translations/fr';
import it from './translations/it';
import pt from './translations/pt';
import { TranslationKeys } from './translations/translations.interface';

export const getCurrentLocale = () => Localization.locale;

export const DEFAULT_LANGUAGE = 'en';
export const DEFAULT_TRANSLATION = {
  key: DEFAULT_LANGUAGE,
  value: 'English',
};

export const getCurrentLanguage = () => StorageUtils.getItem('USER_LANGUAGE') ?? DEFAULT_LANGUAGE;

export const changeLanguage = (newLocale: string) => {
  i18n.locale = newLocale;
};

export const translate = (key: TranslationKeys): string => i18n.t(key);

export const LANGUAGES = [
  { lang: 'pt', file: pt, name: 'Language_Pt' },
  { lang: 'en', file: en, name: 'Language_En' },
  { lang: 'es', file: es, name: 'Language_Es' },
  { lang: 'it', file: it, name: 'Language_It' },
  { lang: 'de', file: de, name: 'Language_De' },
  { lang: 'fr', file: fr, name: 'Language_Fr' },
];

const setupInitialLanguage = () => {
  i18n.fallbacks = true;

  // Set the key-value pairs for the different languages you want to support.
  i18n.translations = LANGUAGES.reduce((translations, { lang, file }) => {
    translations[lang] = file;
    return translations;
  }, {});

  // Set the locale once at the beginning of your app.
  i18n.locale = getCurrentLocale();
};

setupInitialLanguage();
