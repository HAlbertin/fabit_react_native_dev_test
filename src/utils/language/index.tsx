import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import de from './translations/de';
import en from './translations/en';
import es from './translations/es';
import fr from './translations/fr';
import it from './translations/it';
import pt from './translations/pt';
import { TranslationKeys } from './translations/translations.interface';

const LANGUAGES = [
  { lang: 'pt', file: pt },
  { lang: 'en', file: en },
  { lang: 'es', file: es },
  { lang: 'it', file: it },
  { lang: 'de', file: de },
  { lang: 'fr', file: fr },
];

// Set the key-value pairs for the different languages you want to support.
i18n.translations = LANGUAGES.reduce((translations, { lang, file }) => {
  translations[lang] = file;
  return translations;
}, {});

// Set the locale once at the beginning of your app.
i18n.locale = Localization.locale;

export const changeLanguage = (newLocale: string) => {
  i18n.locale = newLocale;
};

export const translate = (key: TranslationKeys): string => i18n.t(key);
