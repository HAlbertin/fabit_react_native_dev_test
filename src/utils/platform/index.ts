import { Platform } from 'react-native';

const isAndroid = () => Platform.OS === 'android';
const isIos = () => Platform.OS === 'ios';

const IOS = '3';
const ANDROID = '2';
const getPlatform = () => (PlatformUtils.isIos ? IOS : ANDROID);

const PlatformUtils = {
  isAndroid,
  isIos,
  getPlatform,
};

export default PlatformUtils;
