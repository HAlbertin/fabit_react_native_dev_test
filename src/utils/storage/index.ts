import { MMKV } from 'react-native-mmkv';
import StorageConstants from '../../constants/storage.constants';

const storage = new MMKV();

type storageKeys = keyof typeof StorageConstants;

const setItem = (keyName: storageKeys, value: string | number | boolean) => {
  storage.set(StorageConstants[keyName], value);
};

const getItem = (keyName: storageKeys) => storage.getString(StorageConstants[keyName]);

const deleteItem = (keyName: storageKeys) => storage.delete(keyName);

const StorageUtils = {
  setItem,
  getItem,
  deleteItem,
};

export default StorageUtils;
