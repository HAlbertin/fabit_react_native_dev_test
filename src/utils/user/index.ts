import StorageUtils from '../storage';

const isUserLoggedIn = () => !!StorageUtils.getItem('USER_SECRET');
const userNeedsVerify = () => !!StorageUtils.getItem('USER_SESSION_VERIFY');

const UserUtils = {
  isUserLoggedIn,
  userNeedsVerify,
};

export default UserUtils;
