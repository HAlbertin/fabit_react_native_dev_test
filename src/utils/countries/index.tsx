import countries from 'i18n-iso-countries';
import de from 'i18n-iso-countries/langs/de.json';
import en from 'i18n-iso-countries/langs/en.json';
import es from 'i18n-iso-countries/langs/es.json';
import fr from 'i18n-iso-countries/langs/fr.json';
import it from 'i18n-iso-countries/langs/it.json';
import pt from 'i18n-iso-countries/langs/pt.json';
import Flag from '../../components/flag';
import { getCurrentLanguage } from '../language';

const __setConfig = () => {
  countries.registerLocale(en);
  countries.registerLocale(es);
  countries.registerLocale(pt);
  countries.registerLocale(it);
  countries.registerLocale(fr);
  countries.registerLocale(de);
};

const GetAllCountries = () => {
  const countriesObj = countries.getNames(getCurrentLanguage(), { select: 'official' });
  const countriesList = Object.entries(countriesObj).map(e => ({
    key: e[0],
    value: e[1],
    icon: <Flag isoCode={e[0]} />,
  }));
  return countriesList;
};

__setConfig();

const CountriesUtils = {
  GetAllCountries,
};

export default CountriesUtils;
