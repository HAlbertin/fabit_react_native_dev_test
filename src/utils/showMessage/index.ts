import { Alert } from 'react-native';
import { translate } from '../language';
import { TranslationKeys } from '../language/translations/translations.interface';

const showMessage = (title: TranslationKeys, message?: TranslationKeys) =>
  Alert.alert(translate(title), message ? translate(message) : '');

const ShowMessageUtils = {
  showMessage,
};

export default ShowMessageUtils;
