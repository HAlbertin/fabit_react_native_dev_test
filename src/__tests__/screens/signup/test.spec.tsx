import { act, fireEvent } from '@testing-library/react-native';
import SignupScreen from '../../../screens/signup';
import { renderWithNavigator } from '../../../utils/test';

const mockUSA_COUNTRY_CODEGetter = jest.fn();
jest.mock('../../../constants/usaStates.constants', () => ({
  get USA_COUNTRY_CODE() {
    return mockUSA_COUNTRY_CODEGetter();
  },
  get US_STATES() {
    return [
      { value: 'ALABAMA', key: 'AL' },
      { value: 'ALASKA', key: 'AK' },
    ];
  },
}));

describe('<SignupScreen />', () => {
  it('should match snapshot', () => {
    const { toJSON } = renderWithNavigator(SignupScreen);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should select a country', async () => {
    const { queryAllByTestId, getByTestId, getByText } = renderWithNavigator(SignupScreen);

    const input = queryAllByTestId('input-select-container')[0];
    await act(async () => {
      await fireEvent.press(input);
    });

    const selectedCountry = getByTestId('list-item-AR');
    await act(async () => {
      await fireEvent.press(selectedCountry);
    });

    const arLabel = getByText('Argentina');

    expect(arLabel).toBeTruthy();
  });

  it('should select a state when US is selected', async () => {
    mockUSA_COUNTRY_CODEGetter.mockReturnValue('AR');
    const { queryAllByTestId, getByTestId, getByText } = renderWithNavigator(SignupScreen);

    const input = queryAllByTestId('input-select-container')[0];
    await act(async () => {
      await fireEvent.press(input);
    });

    const selectedCountry = getByTestId('list-item-AR');
    await act(async () => {
      await fireEvent.press(selectedCountry);
    });

    const arLabel = getByText('Argentina');
    expect(arLabel).toBeTruthy();

    const selectedState = queryAllByTestId('input-select-container')[1];
    await act(async () => {
      await fireEvent.press(selectedState);
    });

    const stateLabel = getByText('ALABAMA');
    expect(stateLabel).toBeTruthy();
  });
});
