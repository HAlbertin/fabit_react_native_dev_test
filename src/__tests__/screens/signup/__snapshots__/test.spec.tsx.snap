// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupScreen /> should match snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Your new account"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RNCSafeAreaView
            edges={
              Array [
                "right",
                "left",
                "bottom",
              ]
            }
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "marginBottom": 28,
                    "marginTop": 10,
                    "paddingBottom": 0,
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "paddingTop": 0,
                  },
                ]
              }
            >
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "marginBottom": 29,
                        "paddingBottom": 0,
                        "paddingLeft": 24,
                        "paddingRight": 24,
                        "paddingTop": 0,
                      },
                    ]
                  }
                >
                  <Text
                    center={true}
                    color="#707070"
                    style={
                      Array [
                        Object {
                          "color": "#707070",
                          "fontSize": 16,
                          "textAlign": "center",
                        },
                      ]
                    }
                    testID="signup-subtitle"
                  >
                    Bacon ipsum dolor amet kielbasa filet mignon biltong hamburger tri-tip sirloin.
                  </Text>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                      "width": "100%",
                    }
                  }
                  testID="input-select-container"
                >
                  <Text
                    color="#707070"
                    style={
                      Array [
                        Object {
                          "color": "#707070",
                          "fontSize": 16,
                        },
                      ]
                    }
                    testID="input-select-label"
                  >
                    What country do you live in?
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderColor": "#CCCCCC",
                          "borderRadius": 4,
                          "borderWidth": 1,
                          "flexDirection": "row",
                          "height": 58,
                          "paddingBottom": 16,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 16,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "width": "95%",
                          },
                        ]
                      }
                    >
                      <Text
                        bold={true}
                        color="#707070"
                        style={
                          Array [
                            Object {
                              "color": "#707070",
                              "fontSize": 16,
                              "fontWeight": "bold",
                            },
                          ]
                        }
                        testID="input-select-text"
                      >
                        Select country
                      </Text>
                    </View>
                    <
                      height={16}
                      style={
                        Array [
                          Object {},
                        ]
                      }
                      width={10}
                    />
                  </View>
                </View>
                <View
                  safeArea={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderTopLeftRadius": 16,
                        "borderTopRightRadius": 16,
                        "elevation": 2,
                        "height": 1284,
                        "position": "absolute",
                        "shadowOpacity": 0.2,
                        "width": "100%",
                      },
                      Object {
                        "bottom": -1334,
                      },
                    ]
                  }
                  testID="bottomsheet-animated-container"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#F7F8F7",
                          "borderRadius": 100,
                          "height": 6,
                          "marginBottom": 10,
                          "marginLeft": "auto",
                          "marginRight": "auto",
                          "marginTop": 10,
                          "width": 82,
                        },
                      ]
                    }
                    testID="bottomsheet-animated-header"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "marginTop": 15,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                        "width": "100%",
                      }
                    }
                    testID="input-select-container"
                  >
                    <Text
                      color="#707070"
                      style={
                        Array [
                          Object {
                            "color": "#707070",
                            "fontSize": 16,
                          },
                        ]
                      }
                      testID="input-select-label"
                    >
                      Which state do you live in?
                    </Text>
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "borderColor": "#CCCCCC",
                            "borderRadius": 4,
                            "borderWidth": 1,
                            "flexDirection": "row",
                            "height": 58,
                            "paddingBottom": 16,
                            "paddingLeft": 16,
                            "paddingRight": 16,
                            "paddingTop": 16,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "width": "95%",
                            },
                          ]
                        }
                      >
                        <Text
                          bold={true}
                          color="#707070"
                          style={
                            Array [
                              Object {
                                "color": "#707070",
                                "fontSize": 16,
                                "fontWeight": "bold",
                              },
                            ]
                          }
                          testID="input-select-text"
                        >
                          State
                        </Text>
                      </View>
                      <
                        height={16}
                        style={
                          Array [
                            Object {},
                          ]
                        }
                        width={10}
                      />
                    </View>
                  </View>
                  <View
                    safeArea={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderTopLeftRadius": 16,
                          "borderTopRightRadius": 16,
                          "elevation": 2,
                          "height": 1284,
                          "position": "absolute",
                          "shadowOpacity": 0.2,
                          "width": "100%",
                        },
                        Object {
                          "bottom": -1334,
                        },
                      ]
                    }
                    testID="bottomsheet-animated-container"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#F7F8F7",
                            "borderRadius": 100,
                            "height": 6,
                            "marginBottom": 10,
                            "marginLeft": "auto",
                            "marginRight": "auto",
                            "marginTop": 10,
                            "width": 82,
                          },
                        ]
                      }
                      testID="bottomsheet-animated-header"
                    />
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.12)",
                    "borderColor": "transparent",
                    "borderRadius": 0,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "elevation": 0,
                    "minWidth": 64,
                    "shadowColor": "#000000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 4,
                      },
                    ]
                  }
                  testID="button-contained"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        Object {
                          "backgroundColor": "#999999",
                          "height": 64,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "letterSpacing": 1,
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            undefined,
                            false,
                            Object {
                              "color": "rgba(0, 0, 0, 0.32)",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": "500",
                            },
                            Object {
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": "500",
                            },
                            Object {
                              "color": "#101820",
                              "fontWeight": "bold",
                            },
                          ],
                        ]
                      }
                    >
                      <Text
                        color="#101820"
                        style={
                          Array [
                            Object {
                              "color": "#101820",
                              "fontSize": 16,
                            },
                          ]
                        }
                        testID="button-contained-text"
                      >
                        Continue
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RNCSafeAreaView>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</View>
`;
